some initial framework for comments/improvements

RichardSmedley opened the pull request:
[Preview credential](https://confluence.issues.couchbase.com/wiki/spaces/DOCS/pages/1971224949/Docs+Team+demo+site+passwords)

Preview: [URL](https://preview.docs-test.couchbase.com/docs-couchbase-lite-multipeer-sync-3.3/couchbase-lite/current/cbl-whatsnew.html)

RichardSmedley committed:
some initial framework for comments/improvements

RichardSmedley committed:
Update gs-prereqs.adoc

RichardSmedley committed:
recommended version?

RichardSmedley committed:
Merge branch 'multipeer-sync-3.3' of https://github.com/couchbase/docs-couchbase-lite into multipeer-sync-3.3

RichardSmedley commented on modules/android/pages/p2psync-multipeer.adoc:
```diff
@@ -8,3 +8,257 @@ ifdef::show_edition[:page-edition: Enterprise Edition]
+
+=== Supported Platforms
+
+For Android, we recommend using a recent release -- preferably supporting API 34, or more recent -- but older versions should work with the multipeer sync feature.
```
I've kept something in about compat and versions -- let me know if you want it removed entirely for now.

idulo commented on modules/android/pages/gs-prereqs.adoc:
```diff
@@ -70,6 +70,8 @@ To verify whether your device supports the AVX2 instructions set, https://www.in
 
+// more applicable for bluetooth implementations - commenting out for now
+////
 == Supported Versions for Multipeer Replicator
```
Think we should call it Multipeer P2P Replicator so it can be clear that it is not part of our cloud to edge sync, e.g. to SGW. Especially since now it is at the same level as the rest of replicators.

idulo replied:
@RichardSmedley overview suggestion:

The MultipeerReplicator enables lightweight, self-organizing mesh networks for apps running on the same local Wi-Fi. This approach requires minimal setup and automates peer discovery and connectivity management, making it simpler than active-passive P2P configurations (link to active passive).

idulo replied:
Currently, we have both replicators active-passive and multipeer somewhat disjoint, we should be cross referencing. For example, in Active-Passive, have if you want to reap benefits x go to multipeer replicator (this applies to Android and iOS).

idulo replied:
Advantages:

- Auto-discovery over local Wi-Fi (via DNS-SD)
- Lightweight and low-maintenance configuration
- Dynamic mesh topology for optimal peer connectivity
- Secure communication via TLS and certificate-based authentication

idulo commented on modules/android/pages/p2psync-multipeer.adoc:
```diff
@@ -8,3 +8,257 @@ ifdef::show_edition[:page-edition: Enterprise Edition]
+
+
+TIP: Multipeer Replication is currently available for use over IP networks.
+If you are working with a bluetooth network, you will need to use Couchbase Lite's xref::p2psync-websocket.adoc[Active-Passive peer-to-peer].
```
That's not the case, they need to use the Custom Listener, Active-Passive does not support Bluetooth. Please capitalise.

idulo commented on modules/android/pages/p2psync-multipeer.adoc:
```diff
@@ -8,3 +8,257 @@ ifdef::show_edition[:page-edition: Enterprise Edition]
+== Introduction
+// tag::introduction-full[]
+// tag::introduction[]
+Couchbase Lite's Peer-to-Peer synchronization solution offers secure storage and bidirectional data synchronization between edge devices without needing a centralized cloud-based control point.
```
mobile and IoT devices?

RichardSmedley committed:
interim comment incorporation

RichardSmedley committed:
Advantages

RichardSmedley committed:
initial uml diagram

RichardSmedley commented on modules/android/pages/p2psync-multipeer.adoc:
```diff
@@ -1,10 +1,291 @@
-:description: MultipeerReplicator is a peer-to-peer replicator that can advertise its presence, discover other peers, establish connections, and replicate data with those peers.
-
+:description: The Multipeer P2P Replicator enables lightweight, self-organizing mesh networks for apps running on the same local Wi-Fi.
 :source-language: Java
```
Working on the iOS version of this now, with a code sample from Pasin - will upload a draft tomorrow.

RichardSmedley committed:
Merge branch 'release/3.3' into multipeer-sync-3.3

RichardSmedley committed:
interim halt - moving work to Swift pages

RichardSmedley committed:
from Androided doc, added mesh note

RichardSmedley committed:
Configuration

RichardSmedley committed:
Filling out example text

RichardSmedley committed:
Active-passive

RichardSmedley committed:
Tidy-up

RichardSmedley committed:
Android --> Swift

RichardSmedley committed:
mermaid --> graphviz

RichardSmedley committed:
diagram is fixed

RichardSmedley committed:
no patches yet

RichardSmedley committed:
compat updates 3.3

RichardSmedley committed:
comment out (fix)

RichardSmedley committed:
comment out (fix)

RichardSmedley committed:
column fix

RichardSmedley committed:
3.3 release

RichardSmedley committed:
dot O

RichardSmedley committed:
3.3 updates

RichardSmedley committed:
Java 3.3

RichardSmedley committed:
ObjC 3.3

RichardSmedley committed:
3.3 Android

RichardSmedley committed:
comment out and point to iOS

RichardSmedley committed:
--> Swift

RichardSmedley committed:
--> 3.3

RichardSmedley committed:
table fix

RichardSmedley committed:
Deprecation

RichardSmedley committed:
table fix

RichardSmedley committed:
table fix

pasin committed:
Add swift and objective-c code snippets for MultipeerReplicator (#988)

* The code snippet file names are MultipeerReplicator.swift (Swift) and MultipeerReplicator.m (Objective-C)

pasin committed:
Update iOS code snippet (#991)

* Updated peerIdentity codesnippet to delete the expired identity before creating a new one.

* Added multipeer-logdomain snippet.

* Fixed Jenkinsfile

iamfortune committed:
Update page structure (#989)

* restructured page structure and headers

* remove duplicate

* remove duplicate

* restructured headers

* updated restructure headers

* update jenkensfile

iamfortune committed:
multipeer sync for kotlin android (#994)

* multipeer sync for kotlin android

* update formatting

* add content for config

* update the example code path

iamfortune committed:
update include code samples tag

iamfortune committed:
updated to reflect swift code

pasin committed:
Update iOS's authenticatorWithRootCerts() and peerIdentity() code snippet (#998)

* Update authenticatorWithRootCerts and peerIdentity code snippet

Updated authenticatorWithRootCerts and peerIdentity code snippet according to the API changes.

* Fix iOS version in Jenkinsfile

iamfortune committed:
Multipeer sync p2p docs update (#997)

* updated transport modes

* added what's new and updated android sync

* fix style guide

* remove older versions

* updated texts to reflect new changes

iamfortune committed:
Release notes 3.3 (#999)

* release notes for android cbl 3.3

* release notes ios cbl-3.3

* updated timeline to august from july

pasin committed:
Add code snippet for creating a self-signed identity (#1001)

pasin committed:
Fix bug not deleting the expired identity in objective-c code snippet (#1002)

iamfortune committed:
updated the docs, including tips (#1000)

* updated the docs, including tips

* added Bonjour config requirement

* ventilated prose

* multipeer supprt

* fix xml formatting

* move signed into peer identity

* self signed identity tag

borrrden committed:
Add multipeer examples for Android (#1003)

* Add multipeer examples for Android

As part of this also add in PR validation to make sure that Android snippets remain buildable, and cleanup some warnings in existing snippets (nothing that shows in the docs will be affected, it's just to make real warnings when building more noticeable)

* Use correct shebang

* Download jq if not present

iamfortune committed:
connect code examples, fix tag errors

iamfortune committed:
update release notes

iamfortune committed:
updated swift release note

RichardSmedley commented on modules/ROOT/pages/cbl-whatsnew.adoc:
```diff
@@ -9,236 +9,43 @@ ifdef::show_edition[:page-edition: {release}]
+You should be able to upgrade from 3.2.x to 3.3.x without manual intervention.
 
-== Release 3.2.3 (April 2025)
+== Release 3.3.0 (August 2025)
```
Might be worth changing this to September before merging?

iamfortune replied:
Updated for this and for the release notes too

RichardSmedley commented on modules/android/pages/p2psync-websocket.adoc:
```diff
@@ -20,6 +20,20 @@ Related Content -- https://docs.couchbase.com/mobile/{major}.{minor}.{maintenanc
+[TIP]
+.Multipeer P2P Replicator
+====
+Multipeer P2P Replicator is available for
```
perhaps add a link?
`xref:p2psync-multipeer.adoc[Multipeer P2P Replicator]`

RichardSmedley commented on modules/java/partials/release-notes/couchbase-mobile-java-release-note.3.3.0.adoc:
```diff
@@ -0,0 +1,33 @@
+[#maint-3-2-0]
+== 3.3.0 -- July 2025
```
September?

RichardSmedley commented on modules/objc/partials/release-notes/couchbase-mobile-objc-release-note.3.3.0.adoc:
```diff
@@ -0,0 +1,35 @@
+[#maint-3-3-0]
+== 3.3.0 -- July 2025
```
September

RichardSmedley reviewed (commented):
Some small tweaks to make @iamfortune , otherwise LGTM  :+1:

iamfortune committed:
update release months for whats new + release note

iamfortune committed:
update month to september
